### DISCUSSION TITLE
Spanish Verbs: Infinitive and First Person Forms?

### DISCUSSION THREAD

**Discussion by tajmone** [alan-if/alan-i18n#40] [Library Design]
@Rich15, I noticed that every verb in the library includes in its syntax what I believe is the _infinitive_ form of the verb, as well as the 1st person singular present form. E.g. the `examinar` verb is defined as:

```alan
Synonyms
  x, ex, examina, examino, estudio, estudia, estudiar = examinar.

Syntax
  examinar = examinar (obj) *
    Where obj IsA thing
      else "¡No puedes examinar" say the obj. "!"
  examinar = examinar el (obj) *.
```

Usually, IF games are played by directing commands in second person: "take apple", i.e. using the _imperative_ form as if commanding the game protagonist (i.e. the _Hero_ in ALAN's terms).

Although this is the most common form, it's not the only one. Some IF systems (especially in the past) allow using different forms of addressing commands (these are my own definitions, just for clarity sake):

1. **Imperative form** (commanding the hero) — e.g. (I order you to:) "take the apple"
2. **First person affirmation** (stating what you do as player) — e.g. (I now:) "take the apple"
3. **Answering the prompt** (replying to an imaginary question: What's next?) — e.g. (I'm thinking of:) "taking the apple"

These examples in English don't really clarify the point, since (1) and (2) are identical, and (3) would be an unusual for ("take the apple" would sound better also there). But in Italian the three approaches would require different verbs:

|    |       approach       |      example       |          verb form          |
|----|----------------------|--------------------|-----------------------------|
| 1. | Commanding the Hero: | esamina la pizza   | _imperative_ 2nd p.s. (you) |
| 2. | What you do now?     | esamino la pizza   | _present_ 1st p.s. (I)      |
| 3. | What you like to do? | esaminare la pizza | _infinitive_                |

Although in Italian most IF games use the classic approach (1) of using the imperative form, in the earlier games you would also see forms (2) and (3), usually when the game prompt was something like `What next? >`.

So, I'm wonder why the Spanish library has all those different verbs forms (tenses and moods), and if this has something to do with the tradition of text adventures in the Spanish language — which, of course, is something worth taking into account.

Otherwise, I think that preserving anything beside the imperative 2nd p.s. is overkill, both in terms of library maintenance (we'd have to ensure that all verbs correctly cover all forms, consistently), but also a burden on end authors, who would also have to cover all those forms in their own custom verbs.

In the Italian library, I use the infinitive for the syntax and verb _identifier_ (because that's the way we refer to any verb, when mentioning it), but no in the actual syntax that the player can type. So, if it was for me, I'd reduce the above definition of `examinar` to:

```alan
Synonyms
  x, ex, estudia = examinar.

Syntax
  examinar = examina (obj) *
    Where obj IsA thing
      else "¡No puedes examinar" say the obj. "!"
```

> **NOTE** — the last line:
>
>       examinar = examinar el (obj) *.
>
> shouldn't even be there! All definitive article should be declared as synonyms of "noise words" (i.e. English "THE") so that the parser will simply ignore them, since they bring no value to parsing player input:
>
>     Synonyms el, los, las, [etc.] = the.
>
> So that's going to be definitely an improvement we'll be taking care of in the grammar module, which will slim down verbs definitions.
>
> For more info, see:
>
> - [ALAN Italian Wiki » Predefined Player Words]

So, Ricardo, what's your view on keeping the infinitive forms of verbs (_examinar_) and the 1st person form (_examino_)? Does it make sense to keep them, because they are commonly use among the Spanish IF community? Or should we drop them and just keep the 2nd person imperative forms?

Historically, the ALAN community has always used only the imperative form so far, and narration has always been in the present tense. But of course, if Spanish IF has a tradition of its own, I don't see why it shouldn't be honoured.

# References

- _[Writing with Inform]_
    + [§14. Adaptive Text and Responses]

[Writing with Inform]: http://inform7.com/book/WI_1_1.html
[§14. Adaptive Text and Responses]: http://inform7.com/book/WI_14_1.html

[ALAN Italian Wiki » Predefined Player Words]: https://github.com/tajmone/Alan3-Italian/wiki/Predefined-Player-Words

**Comment by Rich15** [comment]
@tajmone, I think using the 1st person is unnecessary, since it's very rare to write a command that way (examino obj).  However, it is better to keep the _infinitive_ form because it's more common to say it this way (examinar obj).  

So I think we could delete the 1st person cases and let just the _infinitive_ and _imperative_ forms.  I could do a quick check on all modules to delete these 1st person forms and add some _infinitive_ or _imperative_ ones if necessary.  

***NOTE:***

> But of course, if Spanish IF has a tradition of its own, I don't see why it shouldn't be honoured.

I should note that I'm not aware of the general practices of the Spanish IF community.  I've played some IF games before, but I haven't been an active member of any IF community (until I got here).  So these recommendations are based of my personal experience and general forms of writing in Spanish.  I hope this isn't a problem.

**Reply by tajmone** [reply]
Maybe before deleting anything we should do some research on the the [CAAD] portal and its [WikiCAAD], just to check what the established tradition is. 

If these forms are indeed used we ought to keep them, but if they are rare then I think it would be best to drop them, mostly because they would burden library authors with the duty of having to cover those forms in any custom verb (and deleting the other forms from the library would be even more work for them).

I'd also like to ear @thoni56 on this, for the issue never came up before so we hadn't had a chance to discuss how the new Foundation library should handle this.

From my experience as an IF authors, I've seen that all English IF systems cover narrating in the present tense, and some (e.g. Inform7 and TADS for sure) also allow setting narrative to the past tense. But in all systems commands were only in imperative form (notable exception are "inventory", which isn't a verb, and obviously directions).

I personally think that "less is more", and having a library that does only one thing and does it well is better than having too many features that could end up becoming a burden. But when it comes to gameplay traditions, I believe IF players have very strong expectations, which ultimately determine the way games and libraries are written in each language.

But this was something worth mentioning and discussing. Since we're still in early Beta we have time to think about it and decide just before v1.0.0 is ready (deleting is always easy, restoring what was deleted is not so easy).

I suggest that all library tests and walkthroughs should always use the imperative form, which we're sure will be preserved, so we won't have to tweak the solution files (that would be a true nightmare). Also, if the imperative form works then all the other syntax variations and synonyms surely work too, hence there's no need to test every syntax variation, just the main form is fine.

[CAAD]: https://www.caad.es/
[WikiCAAD]: https://wiki.caad.es/

**Reply by thoni56** [reply]
> I'd also like to ear @thoni56 on this, for the issue never came up before so we hadn't had a chance to discuss how the new Foundation library should handle this.
> 
> From my experience as an IF authors, I've seen that all English IF systems cover narrating in the present tense, and some (e.g. Inform7 and TADS for sure) also allow setting narrative to the past tense. But in all systems commands were only in imperative form (notable exception are "inventory", which isn't a verb, and obviously directions).
> 
> I personally think that "less is more", and having a library that does only one thing and does it well is better than having too many features that could end up becoming a burden. But when it comes to gameplay traditions, I believe IF players have very strong expectations, which ultimately determine the way games and libraries are written in each language.

I agree with you @tajmone. I really think the Foundation library should focus on foundational features, and have good and easily understandable support for those. So we need to be clear about what those are, and that going beyond those is not supported.

And I think that other "forms" than the imperative when it comes to player commands, is outside of the foundational. Perhaps we might have some documentation about how to extend the Foundation library with that. And any current thoughts could be captured on a budding wiki page on the subject.

> But this was something worth mentioning and discussing. Since we're still in early Beta we have time to think about it and decide just before v1.0.0 is ready (deleting is always easy, restoring what was deleted is not so easy).
> 
> I suggest that all library tests and walkthroughs should always use the imperative form, which we're sure will be preserved, so we won't have to tweak the solution files (that would be a true nightmare). Also, if the imperative form works then all the other syntax variations and synonyms surely work too, hence there's no need to test every syntax variation, just the main form is fine.

That is a good approach.

**Reply by Rich15** [reply]
@tajmone @thoni56, after doing some research in the [WikiCAAD](https://wiki.caad.es) and playing several IF adventures in Spanish, I've noticed that most games use both _imperative_ and _infinitive_ form for commands (actually, all of the games I played did it this way, but I say "most" because there could be exceptions). 

If you read Spanish IF wikis you'd probably encounter that they say commands are written in imperative (or 2nd person), but I think this is mainly because most documentations are translated from English.  In practice, almost every game (if not all) allow commands to be written in both forms, and a lot of them even use the _infinitive_ when showing tutorials, instructions and previews of the adventure.

So I don't know if it would be better for the Library to already handle this syntax in verbs, or, as you've already said, be more foundational, and let the autors decide whether they prefer one form over another (depending on the narration of the story, for example) or use both.

What is clear is that commands in 1st person are ***very rarely*** used (almost never).

> I suggest that all library tests and walkthroughs should always use the imperative form, which we're sure will be preserved, so we won't have to tweak the solution files (that would be a true nightmare). Also, if the imperative form works then all the other syntax variations and synonyms surely work too, hence there's no need to test every syntax variation, just the main form is fine.

I agree.

**Reply by tajmone** [reply]
@Rich15, thanks for carrying out this research.

If the infinitive form is part of the Spanish IF tradition we should then leave it. We'll get rid of the 1st person commands though, for that seems overkill.

In any case, we can procrastinate this clean-up operation to when we're about to release v1.0.0.
